geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/server/AbstractRedisMemoryStatsIntegrationTest.java

```diff
@@ -70,24 +67,13 @@ public void memoryFragmentationRatio_shouldBeGreaterThanZero() {
 
   @Test
   public void usedMemory_shouldIncrease_givenAdditionalValuesAdded() {
-    Map<String, String> addedData = makeHashMap(100_000, "field", "value");
-
     long initialUsedMemory = Long.parseLong(RedisTestHelper.getInfo(jedis).get(USED_MEMORY));
-
-    jedis.hset(EXISTING_HASH_KEY, addedData);
+    for (int i = 0; i < 500_000; i++) {
```

Comment:
would it be better to do even more than 500,000 but to check after each set if usedMemory is greater than initialUsedMemory. As soon as you see it > you could break out of the loop. This might make it less likely to be flakey.

Reply:
I'm now checking every 50,000 entries up to a max of 1mil. Bailing out as soon as the condition is met.